subsys/metrics/src/main/java/org/commonjava/indy/metrics/conf/IndyMetricsConfig.java

```diff
@@ -0,0 +1,109 @@
+package org.commonjava.indy.metrics.conf;
+
+import org.commonjava.indy.conf.IndyConfigInfo;
+import org.commonjava.web.config.annotation.ConfigName;
+import org.commonjava.web.config.annotation.SectionName;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.io.InputStream;
+
+/**
+ * Created by xiabai on 3/17/17.
+ */
+@SectionName(IndyMetricsConfig.SECTION)
+@ApplicationScoped
+public class IndyMetricsConfig implements IndyConfigInfo {
```

Comment:
This looks good, but there are a couple of improvements I'd like to see to bring it into line with common idioms we use in Indy:

* Add an enable flag, with default value of false. I'd make it a Boolean field and use a ternary expression in isEnabled(). This allows us to change the default value if necessary later, without changing the field type. It also exposes a constant (DEFAULT_ENABLED) that we can reference in tests and elsewhere.
* Inject IndyMetricsConfig into the interceptor, and if `!config.isEnabled()` just invoke without any of the metrics decoration.
* Change the configuration keys here to things like `reporter` and `graphite.hostname` instead of camel case and package-style key names. We're using a configuration file section header to help us provide a namespace for these keys, so we can use simple keys.